src/utils.rs

```diff
@@ -41,3 +41,36 @@ pub fn handle_communication_err(peer_addr: SocketAddr, e: &Error, details: &str)
     );
     let _ = ctx_mut(|c| c.connections.remove(&peer_addr));
 }
+
+#[cfg(test)]
+pub mod testing {
+    use crate::dirs::{Dirs, OverRide};
+    use crate::NodeInfo;
+    use rand::Rng;
+    use std::env;
+    use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+    use std::path::PathBuf;
+
+    pub fn test_dirs() -> Dirs {
+        Dirs::Overide(OverRide::new(&unwrap!(tmp_dir().to_str())))
+    }
+
+    pub fn rand_peer() -> NodeInfo {
```

Comment:
In integration-test in this PR a test-peer means a fully constructed `QuicP2p` instance ready to be used. This name is in conflict with that - so better to call this `rand_node_info()`

Also not calling it peer (in the light of what test-peer above does) makes sense as it's not usable really as the random port is chosen by you not the OS (not problems like getting in a range where permission is denied or already occupied etc,) so might not be useful for creating an actual peer.